core/src/main/java/org/infinispan/transaction/xa/recovery/RecoveryManagerImpl.java

```diff
@@ -128,38 +129,43 @@ public RecoveryIterator getPreparedTransactionsFromCluster() {
    }
 
    @Override
-   public void removeRecoveryInformation(Collection<Address> lockOwners, Xid xid, boolean sync, GlobalTransaction gtx,
+   public CompletionStage<Void> removeRecoveryInformation(Collection<Address> lockOwners, Xid xid, GlobalTransaction gtx,
                                          boolean fromCluster) {
       log.tracef("Forgetting tx information for %s", gtx);
       //todo make sure this gets broad casted or at least flushed
       if (rpcManager != null && !fromCluster) {
          TxCompletionNotificationCommand ftc = commandFactory.buildTxCompletionNotificationCommand(xid, gtx);
-         sendTxCompletionNotification(lockOwners, ftc, sync);
+         removeRecoveryInformation(xid);
+         return sendTxCompletionNotification(lockOwners, ftc);
+      } else {
+         removeRecoveryInformation(xid);
+         return CompletableFutures.completedNull();
       }
-      removeRecoveryInformation(xid);
    }
 
    @Override
-   public void removeRecoveryInformationFromCluster(Collection<Address> where, long internalId, boolean sync) {
+   public CompletionStage<Void> removeRecoveryInformationFromCluster(Collection<Address> where, long internalId) {
       if (rpcManager != null) {
          TxCompletionNotificationCommand ftc = commandFactory.buildTxCompletionNotificationCommand(internalId);
-         sendTxCompletionNotification(where, ftc, sync);
+         CompletionStage<Void> stage = sendTxCompletionNotification(where, ftc);
+         return stage.thenRun(() -> removeRecoveryInformation(internalId));
```

Comment:
Don't need to wait for the tx completion responses here either.

Reply:
Okay.